test/integration/proxy-config.test.js

```diff
@@ -0,0 +1,36 @@
+var expect = require('expect.js'),
+    ProxyConfig = require('../../lib/index.js').ProxyConfig;
+
+/* global describe, it */
+describe('Proxy Config', function () {
```

Comment:
Should add a lot of tests for the `ProxyConfig.test` method specifically. In fact, we should have 100% branch coverage for that method.